process/process.go

```diff
@@ -146,7 +148,14 @@ func parseMappings(mapsFile io.Reader) ([]Mapping, error) {
 			}
 		} else {
 			path = trimMappingPath(path)
-			path = strings.Clone(path)
+			if path == lastPath {
+				// Take advantage of the fact that mappings are sorted by path
+				// and avoid allocating the same string multiple times.
+				path = lastPath
+			} else {
```

Comment:
Can we skip the assignment of `path = lastPath` if `path == lastPath`?
```suggestion
			if path != lastPath {
```

Reply:
> Can we skip the assignment of path = lastPath if path == lastPath?

No, this is exactly where the de-duplication happens.
`path == lastPath` compares the string content, and here both contents are at different memory locations. So if the contents are the same, the `path = lastPath` changes the memory location of `path` to the already existing `lastPath`.